src/snowflake/cli/plugins/snowpark/package_utils.py

```diff
@@ -130,16 +87,23 @@ def install_packages(
     which are available on the Snowflake Anaconda channel. These will have
     been deleted from the local packages folder.
     """
+    if file_name and package_name:
+        raise ClickException(
+            "Could not use package name and requirements file simultaneously"
+        )
+
+    if file_name and not Path(file_name).exists():
+        raise ClickException(f"File {file_name} does not exists.")
+
     with Venv() as v:
-        if file_name is not None:
-            pip_install_result = v.pip_install(file_name, RequirementType.FILE)
-            dependencies = v.get_package_dependencies(file_name, RequirementType.FILE)
-
-        if package_name is not None:
-            pip_install_result = v.pip_install(package_name, RequirementType.PACKAGE)
-            dependencies = v.get_package_dependencies(
-                package_name, RequirementType.PACKAGE
-            )
+        if package_name:
+            # This is a Windows workaround where use TemporaryDirectory instead of NamedTemporaryFile
+            tmp_requirements = Path(v.directory.name) / "requirements.txt"
+            tmp_requirements.write_text(str(package_name))
+            file_name = str(tmp_requirements)
```

Comment:
I love the idea, it really simplifies logic. But maybe let's move it to a separate function, for readability purposes?